core/src/main/java/org/apache/iceberg/rest/responses/FetchScanTasksResponseParser.java

```diff
@@ -43,7 +43,9 @@ public static String toJson(FetchScanTasksResponse response, boolean pretty) {
   public static void toJson(FetchScanTasksResponse response, JsonGenerator gen) throws IOException {
     Preconditions.checkArgument(response != null, "Invalid response: fetchScanTasksResponse null");
     Preconditions.checkArgument(
-        response.specsById() != null, "Cannot serialize fetchScanTasksResponse without specsById");
+        response.specsById() != null
+            || (response.fileScanTasks() == null || response.fileScanTasks().isEmpty()),
```

Comment:
I fixed the condition for failing when specs by id is missing as well. We only need to have specs by Id defined when the response has file scan tasks, but it's not needed if there's just plan tasks.

Reply:
Nice catch thanks!